### PR TITLE
switch to using featureEnabled for enterprise features

### PR DESCRIPTION
**Pull request by DanCech**
This PR changes the way we validate license entitlements so that we can check for specific features.

### CODE REVIEW

File: public/app/features/admin/state/actions.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ export function loadAdminUserPage(userId: number): ThunkResult<void> {
       await dispatch(loadUserProfile(userId));
       await dispatch(loadUserOrgs(userId));
       await dispatch(loadUserSessions(userId));
-      if (config.ldapEnabled && config.licenseInfo.hasLicense) {
+      if (config.ldapEnabled && featureEnabled('ldapsync')) {
```
Comment (marefr) [inline public/app/features/admin/state/actions.ts:38]: Nit. In general I think that maybe the backend should provide to the frontend if ldapsync is enabled based on licensing and whether ldap is acually enabled to get rid of checking multiple flags in frontend. But probably a bigger change required.

File: public/app/features/datasources/state/navModel.ts [hunk 2]
```diff
@@ -47,7 +48,7 @@ export function buildNavModel(dataSource: DataSourceSettings, plugin: GenericDat
     });
   }
 
-  if (config.licenseInfo.hasLicense) {
+  if (featureEnabled('dspermissions')) {
```
Comment (marefr) [inline public/app/features/datasources/state/navModel.ts:51]: Nit. Maybe move all these "magic" strings to constants and/or a separate file?

**Review by marefr — CHANGES_REQUESTED** [review]
LGTM. Seems to be one thing related to `AppNameBodyClass` blocking from getting this merge. See comments. In general it looks good and makes sense. 

One question is whether these enabled features somehow are or should be related to feature toggles? Not really an issue right now, but there are plans for improving feature toggles framework.

File: pkg/api/index.go [hunk 3]
```diff
@@ -582,7 +582,7 @@ func (hs *HTTPServer) setIndexViewData(c *models.ReqContext) (*dtos.IndexViewDat
 		NewGrafanaVersion:       hs.updateChecker.LatestGrafanaVersion(),
 		NewGrafanaVersionExists: hs.updateChecker.GrafanaUpdateAvailable(),
 		AppName:                 setting.ApplicationName,
-		AppNameBodyClass:        getAppNameBodyClass(hs.License.HasValidLicense()),
+		AppNameBodyClass:        "app-grafana",
```
[the remark this answers is not in the document]
Comment (DanCech) [reply pkg/api/index.go:585]: I realized that the switch can be done in the enterprise codebase, so there's no need for the added complexity here.

File: public/app/features/admin/state/actions.ts (hunk 1 above)
Comment (DanCech) [reply public/app/features/admin/state/actions.ts:38]: it's a good comment, and I had the same thought regarding the relationship between features being enabled via config, feature toggles and/or licensing.  This PR has already grown quite large so I figure we can figure out whether it makes sense to combine them as a separate issue.

File: public/app/features/datasources/state/navModel.ts (hunk 2 above)
Comment (DanCech) [reply public/app/features/datasources/state/navModel.ts:51]: I thought about that also, right now the way we treat feature toggles is similar eg `config.featureToggles['accesscontrol']` so I didn't go the route of trying to define the list centrally.  We could certainly define them as constants in `licensing.ts` if we think there's a benefit to doing that, though that might be something to think about if we do a refactor to unify with feature toggles.

File: public/app/features/admin/state/actions.ts (hunk 1 above)
Comment (marefr) [reply public/app/features/admin/state/actions.ts:38]: Great. Make sense

File: public/app/features/datasources/state/navModel.ts (hunk 2 above)
Comment (marefr) [reply public/app/features/datasources/state/navModel.ts:51]: Sure make sense

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**